maven-embedder/src/main/java/org/apache/maven/cli/transfer/ConsoleMavenTransferListener.java

```diff
@@ -64,20 +65,36 @@ public synchronized void transferProgressed(TransferEvent event) throws Transfer
         TransferResource resource = event.getResource();
         transfers.put(resource, event.getTransferredBytes());
 
-        StringBuilder buffer = new StringBuilder(128);
         buffer.append("Progress (").append(transfers.size()).append("): ");
 
-        synchronized (transfers) {
-            Iterator<Map.Entry<TransferResource, Long>> entries =
-                    transfers.entrySet().iterator();
-            while (entries.hasNext()) {
-                Map.Entry<TransferResource, Long> entry = entries.next();
-                long total = entry.getKey().getContentLength();
-                Long complete = entry.getValue();
-                buffer.append(getStatus(entry.getKey().getResourceName(), complete, total));
-                if (entries.hasNext()) {
-                    buffer.append(" | ");
+        Iterator<Map.Entry<TransferResource, Long>> entries =
+                transfers.entrySet().iterator();
+        while (entries.hasNext()) {
+            Map.Entry<TransferResource, Long> entry = entries.next();
+            long total = entry.getKey().getContentLength();
+            Long complete = entry.getValue();
+
+            String resourceName = entry.getKey().getResourceName();
+
+            if (printResourceNames) {
+                int idx = resourceName.lastIndexOf('/');
+
+                if (idx < 0) {
+                    buffer.append(resourceName);
+                } else {
+                    buffer.append(resourceName, idx + 1, resourceName.length());
                 }
+                buffer.append(" (");
+            }
+
+            buffer.append(format.formatProgress(complete, total));
```

Comment:
I wonder if the `FileSizeFormat` should be refactored to be given a `StringBuffer` to write into, instead of (or in addition to) the current methods.
But this may be for another PR...

Reply:
Done in the last commit